### DISCUSSION TITLE
ESPnow callback to the sender/sleeping device

### DISCUSSION THREAD

**Discussion by homonto** [HomeSpan/HomeSpan#496] [Ideas]
I am heavily using ESPnow in my projects (non-Apple) but I also use your amazing library for one of my project that is of course Apple specific.
Since I have implemented a callback from the receiver to the sender device (the sender is the sleeping one), I was wondering if similar could be implemented in HomeSpan.
So the detailed idea is:
1- sender wakes up, measures, sends data to the bridge and goes to sleep
2- but before it goes to sleep, it first waits for bridge to send the command to the sender
3- if the command is i.e. "0" then it goes to sleep ASAP
4- if the command is != "0" then sender would do whatever is programmed, i.e. OTA, reset, Captive Portal or so

Just an idea because bloody HomeKit does not allow to talk to the devices so easily

Thank you

**Comment by HomeSpan** [comment]
A SpanPoint object can be both a sender and a receiver so you should be able to do this today.   After the sender sends data to the bridge, have it go into a loop waiting to hear from the bridge using SpanPoint's `get()`, and then act on whatever the bridge sends back.  You may want to add a time-out in the loop in case for whatever reason it never receives the message.  On the bridge side, after it receives measurement data from the sender, have it send something back using the the same SpanPoint link.

**Reply by homonto** [reply]
that is exactly how I saw it done - thank you - let me try ;-)
